File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/views/home/fragments/contact_fragment/ContactFragment.java
package vng.zalo.tdtai.zalo.zalo.views.home.fragments.contact_fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.material.tabs.TabLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.viewpager.widget.ViewPager;
import vng.zalo.tdtai.zalo.R;

public class ContactFragment extends Fragment implements TabLayout.OnTabSelectedListener {
    private ViewPager subContactViewPager;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_contact, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        subContactViewPager = view.findViewById(R.id.subContactViewPager);
        FragmentStatePagerAdapter contactFragmentAdapter = new ContactFragmentAdapter(getChildFragmentManager());
        subContactViewPager.setAdapter(contactFragmentAdapter);

        TabLayout tabLayout = view.findViewById(R.id.subContactTabLayout);
        subContactViewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
        tabLayout.setOnTabSelectedListener(this);
    }

    @Override
    public void onTabSelected(TabLayout.Tab tab) {
        subContactViewPager.setCurrentItem(tab.getPosition());
    }

    @Override
    public void onTabUnselected(TabLayout.Tab tab) {

    }

    @Override
    public void onTabReselected(TabLayout.Tab tab) {

    }
}

File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/views/home/fragments/contact_fragment/ContactFragment.kt
package vng.zalo.tdtai.zalo.zalo.views.home.fragments.contact_fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.google.android.material.tabs.TabLayout
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import kotlinx.android.synthetic.main.fragment_contact.*
import vng.zalo.tdtai.zalo.R

class ContactFragment : Fragment(), TabLayout.OnTabSelectedListener {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_contact, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        initView()
    }

    private fun initView(){
        viewPager.apply {
            adapter= ContactFragmentAdapter(childFragmentManager)
            addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabLayout))
        }

        tabLayout.addOnTabSelectedListener(this)
    }

    override fun onTabSelected(tab: TabLayout.Tab) {
        viewPager.currentItem = tab.position
    }

    override fun onTabUnselected(tab: TabLayout.Tab) {

    }

    override fun onTabReselected(tab: TabLayout.Tab) {

    }
}